extensions/amp-script/0.1/user-activation-tracker.js

```diff
@@ -80,6 +83,28 @@ export class UserActivationTracker {
     return this.lastActivationTime_;
   }
 
+  /**
+   * @param {!Promise} promise
+   */
+  expandLongTask(promise) {
+    if (!this.isActive()) {
+      return;
+    }
+    this.inLongTask_ = true;
+    promise.catch(() => {}).then(() => {
```

Comment:
`promise.catch(() => {}).then(...)` is interesting. Different behavior vs. `Promise.resolve().then(...)`?

Reply:
Yeah. This is basically a DYI `finally()`.